youtube_dl/extractor/bandcamp.py

```diff
@@ -77,6 +86,12 @@ def _real_extract(self, url):
                     'title': data['title'],
                     'formats': formats,
                     'duration': float_or_none(data.get('duration')),
+                    'track': data.get('title'),
+                    'track_number': data.get('track_num'),
+                    'track_id': track_id,
+                    'album': album,
+                    'album_artist': album_artist,
```

Comment:
"album_artist: List of all artists appeared on the album"
Based on that definition, I guess you're right.  I was thinking album artist vs. track artist (like on a mixtape, or similar).